src/seahawk_deck/seahawk_deck/motion_controller.py

```diff
@@ -18,43 +19,119 @@ def __init__(self):
         """Initialize this node"""
         super().__init__('motion_controller')
         self.twist_pub = self.create_publisher(Twist, 'drive/twist', 10)
-        self.motor_pub = self.create_publisher(Int16MultiArray, 'drive/motors', 10) # subscription to this in rov/i2c_proxy
+        self.motor_pub = self.create_publisher(Float32MultiArray, 'drive/motors', 10)
         self.subscription = self.create_subscription(Joy, 'joy', self._callback, 10)
+    
+    def thrust_control(self, direction1:float, direction2:float)->float:
+        """Add two directions in such a way that they do not fall outside [-1, 1]"""
+        if (direction1 >= 0 and direction2 >= 0):
+            return direction1 + direction2 - (direction1 * direction2)
+        elif (direction1 < 0 and direction2 < 0):
+            return direction1 + direction2 + (direction1 * direction2)
+        else:
+            return direction1 + direction2
 
     def _callback(self, joy_msg):
-        """
-        Called every time the joystick publishes a message. 
-        """
+        """Called every time the joystick publishes a message. """
         self.get_logger().info(f"Joystick axes: {joy_msg.axes} buttons: {joy_msg.buttons}")
 
         # Compute desired motion in <x, y, z, r, p, y>
+
+        # CONTROLLER KEYMAPPINGS
+        controller = {
+            'axes': ['left_stick', 'right_stick'],
+            'left_stick': {
+                'x':        -joy_msg.axes[0],
+                'y':        joy_msg.axes[1],
+                'press':    joy_msg.buttons[9],
+            },
+            'right_stick': {
+                'x':        -joy_msg.axes[3],
+                'y':        joy_msg.axes[4],
+                'press':    joy_msg.buttons[10],
+            },
+            'left_trigger': joy_msg.axes[2],
+            'right_trigger':joy_msg.axes[5],
+            'dpad': {
```

Comment:
The dpad of the controller we are testing with is not in fact a list of buttons, even though that would make a lot more sense-- it instead is two axis, each with three states, {-1, 0, 1}. Slightly cursed but if it works it works.

Reply:
I should add that the point of making these assignments in the python dictionary is to normalize them so that in the case we do switch controllers, we can change it out easily enough.

Reply:
We aren't using the dpad for anything right now so I don't think this needs to be resolved here.

We probably don't need to the dictionary to have any indexes not used but I guess this is the only place it's currently documented

Reply:
I'm in favor of simply not mapping the unused controls. It'll be easier to go through and see that it's missing when we need to add it. 

If we keep it, we end up with crusty code that we need to reason about in several weeks or months. A slowdown for no benefit at any time.

Reply:
The benefit is in code readability.

Reply:
The benefit of removal is readability, yeah. :v

To support this: There would be no review process of this vestigial code if it was entirely absent. It's just noise now, and it will be again when we actually go to use the dpad.

Reply:
But therein lies the question: Should we remove it, or comment it out? I'm in favor of commenting it out, were we to remove it.

Reply:
Remove it. Put in a note with a commit hash and a message that says "see here for soke example code"

We don't want to comment out stuff just for reference. That's how we end up with lots of unmaintained code hanging around because they're "comments" and "don't need maintenance."